### PR TITLE
feat: Add config option to ignore files based on Unix shell-style wildcard pattern matching

### LINKED ISSUE #21: bug: `skip_files` is not OS-agnostic
### Description of the bug

The `skip_files` option accepts literal file paths, but when the file is checked it uses OS-specific file path separators, which means each file needs to be specified twice to work on Windows/Unix operating systems.

### To Reproduce

1. Set up mkdocs with plugin on Windows operating system
2. Add a file path to the `skip_files` option using Unix-style file path separators
    ```yaml
       plugins:
         - spellcheck:
             skip_files:
               - path/with/unix/seps/file.txt
     ```
4. Run mkdocs and observe how the file is not skipped

### Expected behavior

The `skip_files` option should accept file paths using Unix-style file path separators and work on both Windows and Unix operating systems.

### Environment information

```console
$ python -m mkdocs_spellcheck.debug
- __System__: Windows-10-10.0.19045-SP0
- __Python__: cpython 3.11.8
- __Environment variables__:
- __Installed packages__:
  - `mkdocs-spellcheck` v1.0.3
```

### PR DESCRIPTION
**Pull request by nfelt14**
This adds the `skip_file_globs` config option to enable skipping files based on a Unix shell-style glob. This also updates the `skip_files` config option to treat file paths as OS-agnostic.

fixes #21 
resolves: #20

### CODE REVIEW

**Comment by nfelt14** [comment]
@pawamoy let me know if there are any changes necessary in this PR.

File: .gitignore [hunk 1]
```diff
@@ -5,6 +5,7 @@
 # python
 *.egg-info/
 *.py[cod]
+pdm.lock
```
Comment (pawamoy) [inline .gitignore:8]: Since this project uses `uv` now, this file shouldn't be created anymore, and you can just delete it if you still have one.

```suggestion
```

File: src/mkdocs_spellcheck/plugin.py [hunk 2]
```diff
@@ -66,6 +67,7 @@ class SpellCheckPlugin(BasePlugin):
         ("backends", MkType(list, default=["symspellpy"])),
         ("known_words", MkType((str, list), default=[])),
         ("skip_files", MkType(list, default=[])),
+        ("skip_file_globs", MkType(list, default=[])),
```
Comment (pawamoy) [inline src/mkdocs_spellcheck/plugin.py:70]: Can't we simply merge the glob feature into the existing `skip_files` option?

File: src/mkdocs_spellcheck/plugin.py [hunk 3]
```diff
@@ -89,7 +91,10 @@ def on_config(self, config: MkDocsConfig) -> MkDocsConfig | None:
         """
         self.strict_only = self.config["strict_only"]
         self.backends_config = self.config["backends"]
-        self.skip_files = self.config["skip_files"]
+        # Convert the skip files into a set of Path objects to make it
+        # OS-agnostic and increase lookup efficiency
+        self.skip_files = {Path(x) for x in self.config["skip_files"]}
```
Comment (pawamoy) [inline src/mkdocs_spellcheck/plugin.py:96]: Could you move this change to another PR, so that we have two different fix and feat commits?

**Review by pawamoy — CHANGES_REQUESTED** [review]
Thanks for the PR!

File: src/mkdocs_spellcheck/plugin.py (hunk 2 above)
Comment (nfelt14) [reply src/mkdocs_spellcheck/plugin.py:70]: That should work. I will see if I can get to it today.

File: src/mkdocs_spellcheck/plugin.py (hunk 3 above)
Comment (nfelt14) [reply src/mkdocs_spellcheck/plugin.py]: I removed this change, but the changes to add fnmatch still fix the issue, so the new feature also fixes the bug.

File: src/mkdocs_spellcheck/plugin.py (hunk 2 above)
Comment (nfelt14) [reply src/mkdocs_spellcheck/plugin.py]: I combined the options.

File: src/mkdocs_spellcheck/plugin.py [hunk 4]
```diff
@@ -129,7 +130,7 @@ def on_page_content(self, html: str, page: Page, **kwargs: Any) -> None:  # noqa
             page: The page instance.
             **kwargs: Additional arguments passed by MkDocs.
         """
-        if self.run and page.file.src_path not in self.skip_files:
+        if self.run and not any(fnmatch.fnmatch(page.file.src_path, pattern) for pattern in self.skip_files):
```
Comment (nfelt14) [inline src/mkdocs_spellcheck/plugin.py:133]: This change both adds the new feature and fixes the bug in a single line.

**Comment by nfelt14** [comment]
> Thanks for the PR!

You are welcome. Let me know if there are any other changes needed.

**Comment by nfelt14** [comment]
> > Thanks for the PR!
> 
> You are welcome. Let me know if there are any other changes needed.

@pawamoy, I think this is ready now.

**Comment by pawamoy** [comment]
Yep, thanks for the notification, I'll review this tomorrow :slightly_smiling_face:

**Review by pawamoy — APPROVED** [review]
OK LGTM, thanks a lot!
